grails-test-suite-uber/src/test/groovy/grails/compiler/DomainClassWithInnerClassUsingStaticCompilationSpec.groovy

```diff
@@ -0,0 +1,60 @@
+package grails.compiler
+
+import grails.compiler.GrailsCompileStatic
+import grails.persistence.Entity
+import grails.testing.gorm.DomainUnitTest
+import grails.validation.Validateable
+import spock.lang.Issue
+import spock.lang.Specification
+
+class DomainClassWithInnerClassUsingStaticCompilationSpec extends Specification implements DomainUnitTest<SomeClass> {
+
+    @Issue('https://github.com/grails/grails-core/issues/12461')
+    void 'a domain class marked with @GrailsCompileStatic containing an inner class and a "constraints" block'() {
+        expect: 'the configuration from the "constraints" closure is available'
+            SomeClass.constraints instanceof Closure
+            SomeClass.constraintsClosureCalled
+    }
+
+    @Issue('https://github.com/grails/grails-core/issues/12461')
+    void 'a domain class marked with @GrailsCompileStatic containing an inner class and a "mapping" block'() {
+        expect: 'the configuration from the "mapping" closure is available'
+            SomeClass.mapping instanceof Closure
+            SomeClass.mappingClosureCalled
+    }
+
+    @Issue('https://github.com/grails/grails-core/issues/12461')
+    void 'a domain class marked with @GrailsCompileStatic containing an inner class and a "namedQueries" block'() {
+        setup:
+            SomeClass.getNamedQuery('test')
```

Comment:
This is required in contrast to the other tests as the `namedQueries` closure seems only to be evaluated when the first named query is accessed and not when the class is compiled.